src/pluto/pluto.py

```diff
@@ -395,17 +396,56 @@ def create_null(df_null, v1, v2, v3, condo, mapped):
                 "pfirm15_flag",
             ]
 
-            def generate_graph(r, total, title):
-                y = [r[i] for i in x]
-                text = [f"{round(r[i]/total*100, 2)}%" for i in x]
-                return go.Scatter(
-                    x=x,
-                    y=y,
-                    mode="lines",
-                    name=title,
-                    hovertemplate="<b>%{x} %{text}</b>",
-                    text=text,
+            field_range = st.slider(
+                "Select a range of values",
+                min_value=0,
+                max_value=len(x),
+                value=[0, 12],
+            )
+
+            def sort_and_filter_df(df, range):
+                df = df[["pair"] + df.columns.tolist()[range[0] + 1 : range[1] + 1]]
+                df_transformed = (
+                    df.set_index("pair")
+                    .stack()
+                    .reset_index()
+                    .rename(
+                        columns={"pair": "version", "level_1": "field", 0: "change"}
+                    )
+                    .sort_values(by="change", ascending=False)
                 )
+                return df_transformed
+
+            def display_graph(df_transformed, range, grouped=True):
+                if grouped:
+                    fig1 = px.bar(
+                        data_frame=df_transformed,
+                        y="change",
+                        x="field",
+                        barmode="group",
+                        color="version",
+                        color_discrete_sequence=COLOR_SCHEME,
+                        height=400,
+                        width=850,
+                        text_auto=True,
+                        title="Null Graph",
+                    )
+                    fig1.update_layout(legend_title_text="Version")
+                else:
+                    fig1 = px.bar(
+                        data_frame=df_transformed,
+                        y="change",
+                        x="field",
+                        color_discrete_sequence=COLOR_SCHEME,
+                        height=400,
+                        width=850,
+                        text_auto=True,
+                        title="Null Graph",
+                    )
```

Comment:
If we are going to re-use so many arguments we could construct a `kwargs` dict and pass that to DRY it out